pkg/tar/tar.go

```diff
@@ -356,13 +356,13 @@ func extractFile(dir string, header *tar.Header, tarReader io.Reader) error {
 	glog.V(3).Infof("Creating %s", path)
 
 	file, err := os.Create(path)
+	if err != nil {
+		return err
+	}
```

Comment:
is the file returned from os.Create considered opened?  If so, is there any condition under which os.Create returns valid open file, and an error? (in which case you are not going to close the file with this logic)